exchanges/stream/websocket_connection.go

```diff
@@ -192,6 +224,9 @@ func (w *WebsocketConnection) IsConnected() bool {
 
 // ReadMessage reads messages, can handle text, gzip and binary
 func (w *WebsocketConnection) ReadMessage() Response {
+	if w.Connection == nil {
+		return Response{}
+	}
```

Comment:
Has this happened? Is there a scenario I've missed?

Reply:
Yeah caused by me and an incorrect configuration. If this is called before dialing it panics.

Reply:
TL;DR: I think we should take the warning labels off the boxes.

I think panicing is the best thing to do there.
Discussed a few times with @thrasher-, and if developer error is the only way to cause something, and it happens during setup/startup, then that's fine.
The failure they'd get is that Connection is nil.
If they've managed to cause that, they'll be able to understand it as well.

Reply:
I'm happier with a middleground. Fix the issue that caused your panic and RM this section. eg have a block on `(w *Websocket) Reader` until `Connected` is true

Reply:
Purged this contentious issue. 😆